apps/passport-server/src/util/telegramWebApp.ts

```diff
@@ -21,33 +21,87 @@ export const closeWebviewHtml = `
     </html>
   `;
 
-export const errorHtmlWithDetails = (error: string): string => {
+export const errorHtmlWithDetails = (error: Error): string => {
+  const errorMessage = error ? error.message.replace(/\n/g, "<br>") : "";
+  const errorStack =
+    error instanceof Error && error.stack
+      ? error.stack.replace(/\n/g, "<br>")
+      : "";
   return `<!DOCTYPE html>
   <html>
-  <head>
-    <title>Error</title>
-    <style>
-    body {
-      background-color: white;
-    }
-    </style>
-  </head>
-  <body>
-  <h3>Action failed</h3>
-  ${
-    error
-      ? `
-      <p>Here is the error we received:</p>
-      <p><b>${error}</b></p>
-      `
-      : ""
-  }
-    <p>Type <i>/start</i> to try again.</p>
-    <p>
-      If you need help additional help, please email
-      <b>passport@0xparc.org</b>.
-    </p>
+    <head>
+      <title>Error</title>
+      <style>
+        body {
+          background-color: white;
+          font-family: -apple-system, BlinkMacSystemFont, avenir next, avenir, segoe ui, helvetica neue, helvetica, Cantarell, Ubuntu, roboto, noto, arial, sans-serif;
+        }
+        .container {
+          margin: 1rem;
+          display: flex;
+          flex-direction: column;
+          gap: 0.8rem;
+        }
+        .header {
+          color: #2a3231;
+          font-size: 1.2rem;
+          font-weight: 500;
+        }
+        .description {
+          color: #585b6b;
+        }
 
+        .toggleButton {
+          cursor: pointer;
+          border: none;
+          color: rgba(0, 0, 0, 0.5);
+          background: #fff;
+          padding: 0.5rem 1rem;
+          border-radius: 0.3rem;
+          font-size: 1rem;
+        }
+
+        .contactButton {
+          cursor: pointer;
+          border: none;
+          color: #fff;
+          background: #2a3231;
+          padding: 0.5rem 1rem;
+          border-radius: 0.3rem;
+          font-size: 1rem; 
+        }
+
+        .errorStack {
+          display: none;
+          color: #de6c5f;
+          word-wrap: break-word;
+        }
+
+        .errorStack.show {
+          display: block;
+        }
+      </style>
+    </head>
+    <body>
+      <div class="container">
+        <span class="header">${error.name}</span>
+        <span class="description">
+          ${errorMessage}
+        </span>
+        <button class="contactButton">Message Zucat Support</button>
+        <button class="toggleButton" onclick="toggleErrorStack()">Show Error Details (Advanced)</button>
```

Comment:
suggestion: It's not super clear that this is expandable. Adding a "+" or a toggle arrow would help